### PR TITLE
Fix SCRAM implant not working while cuffed. Incidentally fix freedom implant working while dead/crit

### LINKED ISSUE #25662: SCRAM implant doesn't work while cuffed
## Description
When attempting to use SCRAM implanter while restrained, you just. won't

**Reproduction**
>be synd
>scram!
>cuffed
>scram!
>still in place

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by nikthechampiongr**
fixes #25662 

## About the PR

This pr makes it so you can use the SCRAM implant while cuffed. Also fixes the freedom implant working while you are in crit, or even dead.

## Why / Balance

The implant is made so you can escape horrible situations. It should as such work while in crit.

## Technical details

ActionBlockers now has a specific check to see if someone is conscious. It is also used in the Interaction check now. Actions can now use that check if the bool is set. The bool is now set for the freedom and SCRAM!  implant actions. Additionally the SCRAM! action now ignores whether the user can interact in order to be able to use it while cuffed.

## Media

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
fix: You can now use the SCRAM! implant while cuffed.
fix: The freedom implant can no longer be used while in crit, or dead.

### CODE REVIEW

**Comment by IProduceWidgets** [comment]
Consider: Triggering automatically until out of charges is a pretty funny drawback.

File: Content.Shared/Actions/SharedActionsSystem.cs [hunk 1]
```diff
@@ -370,6 +371,13 @@ private void OnActionRequest(RequestPerformActionEvent ev, EntitySessionEventArg
 
         BaseActionEvent? performEvent = null;
 
+        if (action.CheckMobState != null)
+        {
+            if (!TryComp<MobStateComponent>(user, out var mobState) ||
+                mobState.CurrentState != action.CheckMobState)
+                return;
+        }
+
```
Comment (metalgearsloth) [inline Content.Shared/Actions/SharedActionsSystem.cs:380]: Could this just use ActionBlocker instead?

Comment (nikthechampiongr) [reply Content.Shared/Actions/SharedActionsSystem.cs:380]: I guess so. I just realized it also needs to check if someone is sleeping. I'll just make a new check named IsConcious or something.

File: Content.Shared/Actions/BaseActionComponent.cs [hunk 2]
```diff
@@ -118,6 +119,12 @@ public abstract partial class BaseActionComponent : Component
     /// </summary>
     [DataField("checkCanInteract")] public bool CheckCanInteract = true;
 
+    /// <summary>
+    /// Whether to check if the user is conscious or not. Can be used instead of <see cref="CheckCanInteract"/>
+    /// for a more permissive check.
+    /// </summary>
+    [DataField] public bool CheckConsciousness;
```
Comment (EmoGarbage404) [inline Content.Shared/Actions/BaseActionComponent.cs:126]: This should be true by default

File: Content.Shared/Mobs/Systems/MobStateSystem.Subscribers.cs [hunk 3]
```diff
@@ -28,7 +28,7 @@ private void SubscribeEvents()
         SubscribeLocalEvent<MobStateComponent, ChangeDirectionAttemptEvent>(CheckAct);
         SubscribeLocalEvent<MobStateComponent, UseAttemptEvent>(CheckAct);
         SubscribeLocalEvent<MobStateComponent, AttackAttemptEvent>(CheckAct);
-        SubscribeLocalEvent<MobStateComponent, InteractionAttemptEvent>(CheckAct);
+        SubscribeLocalEvent<MobStateComponent, ConsciousAttemptEvent>(CheckAct);
```
Comment (EmoGarbage404) [inline Content.Shared/Mobs/Systems/MobStateSystem.Subscribers.cs:31]: Can you also add a check for sleeping players?

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]

File: Content.Shared/Mobs/Systems/MobStateSystem.Subscribers.cs (hunk 3 above)
Comment (nikthechampiongr) [reply Content.Shared/Mobs/Systems/MobStateSystem.Subscribers.cs:31]: Not sure if I misunderstood you, in discord you said the consciousness check should check if the user is asleep. It does that as the sleepsystem subscribes to it so this should be resolved?

**Comment by nikthechampiongr** [comment]
Oh, these test failures are not my fault.

**Review by VasilisThePikachu — APPROVED** [review]

File: Content.Shared/Mobs/Systems/MobStateSystem.Subscribers.cs (hunk 3 above)
Comment (VasilisThePikachu) [reply Content.Shared/Mobs/Systems/MobStateSystem.Subscribers.cs:31]: nik is right this is already checked for sleeping

**Pull request merged**